crates/exex/src/db.rs

```diff
@@ -0,0 +1,343 @@
+use std::{
+    collections::HashMap,
+    str::FromStr,
+    sync::{Arc, Mutex, MutexGuard},
+};
+
+use reth_primitives::{
+    revm_primitives::{AccountInfo, Bytecode},
+    Address, Bytes, SealedBlockWithSenders, StorageEntry, B256, U256,
+};
+use reth_provider::{bundle_state::StorageRevertsIter, OriginalValuesKnown};
+use reth_revm::db::{
+    states::{PlainStorageChangeset, PlainStorageRevert},
+    BundleState,
+};
+use rusqlite::Connection;
+
+/// Types used inside RevertsInit to initialize revms reverts.
+pub type AccountRevertInit = (Option<Option<AccountInfo>>, Vec<StorageEntry>);
+
+/// Type used to initialize revms reverts.
+pub type RevertsInit = HashMap<Address, AccountRevertInit>;
+
+/// A struct representing the database, encapsulating a connection to the SQLite database.
+#[derive(Debug)]
+pub struct Database {
```

Comment:
@greged93 Here I've just used the database because it is used as a state in this example and I think that it is simpler, at least at the beginning to check stuffs, to fetch information, to do assertions on tests, debug...

I see this as a small state in a super simple thing which is a sqlite database.

Reply:
agreed, just fixed